### PR TITLE
Blueshield: Advanced Stun Revolver

### PR DESCRIPTION
**Pull request by DaveTheHeadcrab**
Due to some _very_ vocal requests for Blueshield to have a lethal weapon in the face of the revolver getting nerfed, I have made the advanced stun revolver.

The Blueshield's locker no longer contains the revolver, and instead the Blueshield will spawn with the advanced stun revolver on their person (As a late joiner or if they spawned at round start). It fits in the shoulder holster for easy access in emergencies.

The gun serves the two vital functions that a Blueshield will require to do their job:
 - Stop approaching threats with a switch rather than a timer with the stun mode on the gun, which will fire an electrode functionally identical to one from a stun revolver.
 - Lethally deal with major threats as they appear with the laser mode on the gun.

Icons:
bsgun_taser
bsgun_laser

*Braces for the shitstorm.*

### CODE REVIEW

File: code/modules/projectiles/guns/energy/blueshield.dm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		charge_cost = 2000
+		fire_delay = 15
+
+		projectile_type = "/obj/item/projectile/energy/electrode"
+		fire_sound = 'sound/weapons/Taser.ogg'
+		modifystate = "bsgun_stun"
+
+		user << "<span class = 'info'>You adjust [src.name] to the stun setting.</span>"
+
+	update_icon()
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/blueshield.dm]: need to update the left and right hands.

**Comment by Fox-McCloud** [comment]
PIcs/sprites.

File: code/modules/projectiles/guns/energy/blueshield.dm (hunk 1 above)
Comment (DaveTheHeadcrab) [inline code/modules/projectiles/guns/energy/blueshield.dm]: No I don't. The item_state is simply "gun".

**Comment by Fox-McCloud** [comment]
@Plotron @Markolie 

I know at least those two will care about this.

File: code/modules/projectiles/guns/energy/blueshield.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	item_state = "gun"
+	modifystate = "bsgun_stun"
+	force = 7
+	fire_sound = 'sound/weapons/Taser.ogg'
+	charge_cost = 2000
+	projectile_type = "/obj/item/projectile/energy/electrode"
+	fire_delay = 15
+	var/fire_mode = 0 // 0 for taser, 1 for lethals.
+
+/obj/item/weapon/gun/energy/blueshield/attack_self(mob/living/user as mob)
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/blueshield.dm]: gun code projectile switching makes me want to *scream

**Comment by DaveTheHeadcrab** [comment]
@Fox-McCloud 
Pics in the post now.

File: code/modules/projectiles/guns/energy/blueshield.dm (hunk 1 above)
Comment (tigercat2000) [inline code/modules/projectiles/guns/energy/blueshield.dm]: that isn't really relevant, humans have to have their inhand overlays regenerated

**Comment by NTSAM** [comment]
:ok:

**Comment by HereticChurch** [comment]
I like it. The sprite looks a little bland though. Perhaps using something similar to the sprites made up by Johannhawk on the forums? http://nanotrasen.se/phpBB3/viewtopic.php?f=17&t=3081&start=90#p34617

**Comment by Regen1** [comment]
Agree with @HereticChurch  on the sprite

It's a good midway between instastuns and lethality

**Comment by Plotron** [comment]
Thanks, @DaveTheHeadcrab. I agree with @HereticChurch  and @Regen1, I'd like you to ask Johannhawk on the forums for the sprites - he seems to be good at it. Especially considering that he made some cool variants [here](http://nanotrasen.se/phpBB3/viewtopic.php?f=17&t=3081&start=90#p34650) to pick from.
Is it a shame that the Blueshield will no longer be able to customize the gun's name? I think not. If anything, the Detective's revolver should be unique in that aspect.

Just make sure this energy pistol will spawn on plasmamen when they late-join.

**Comment by Earthdivine** [comment]
:+1: to the idea, Sprite really needs work like already mentioned.

I personally would love for this to be nameable too, it was always interesting to see the gimmick names people could come up with. But if not then that's no big deal.

**Comment by MindLeaker** [comment]
:+1:, I don't think the Blueshield should ever have been equipped with a revolver, so this is welcome change for me.

**Comment by DaveTheHeadcrab** [comment]
Icon updated. check the OP.
@Earthdivine 
@HereticChurch 
@Plotron

**Comment by Spacemanspark** [comment]
:+1: I've suggested Blueshields get an E-gun of sorts, this works better than the revolver.

File: code/modules/projectiles/guns/energy/blueshield.dm (hunk 1 above)
Comment (DaveTheHeadcrab) [inline code/modules/projectiles/guns/energy/blueshield.dm]: It's entirely relevant.
The inhand icon is supposed to be regenerated because it swaps details on the gun's inhand when the mode swaps. 
That is not the case with this gun.

**Comment by HereticChurch** [comment]
:+1: :100:

**Comment by DaveTheHeadcrab** [comment]
Update:
Changed the electrode fire sound to match the stun revolver's due to the new icon.
Gun name and description have been changed accordingly.

File: code/modules/projectiles/guns/energy/blueshield.dm (hunk 1 above)
Comment (tigercat2000) [inline code/modules/projectiles/guns/energy/blueshield.dm]: Oh. I see. It doesn't have inhands.
Yeah that'd do it for not needing an icon update.

**Comment by Plotron** [comment]
@DaveTheHeadcrab, this new icon has much better style now. I'm sure it will look even better in-game.

**Comment by DaveTheHeadcrab** [comment]
Due to reports of plasmaman spawning with the gun being buggy the gun's spawn location has been changed to the backpack of the blueshield of left hand if unavailable.

**Comment by KasparoVy** [comment]
Looks sexy as fuck

Going to miss being able to pen shields but at least it's still looking legit

**Comment by johannhawk** [comment]
i approve.

**Comment by Finalsong** [comment]
:+1: :100:

**Comment by Plotron** [comment]
Not as controversial as you'd think.

**Comment by TheDZD** [comment]
Not sure I like this, introduces another person on the station that can tase-and-lase from roundstart (the other is the HoS, who is generally the most heavily armed and armored crewmember because he is a huge target). The only other way to access tase-and-lase is via advanced eguns, which at least takes some amount of investment, and have the danger of catastrophically failing if they are ever EMPed.

tl;dr More roundstart tase-and-lase access seems like power creep.

**Comment by DaveTheHeadcrab** [comment]
@TheDZD 
It's still objectively less powerful than the old revolver he has currently, which functionally tases a person AND does damage equivocal to a laser in one hit.

**Comment by Fox-McCloud** [comment]
>The only other way to access tase-and-lase is via advanced eguns, which at least takes some amount of investment, and have the danger of catastrophically failing if they are ever EMPed.

Not even; the AeG is laser+disabler, though that may change down the road--but for now, the HoS is literally the only person who has tase and lase.

**Comment by tigercat2000** [comment]
@DaveTheHeadcrab I wouldn't say it's objectively less powerful, revolvers and energy weapons have different uses.

The old revolver, sure, it did the damage of a laser and dropped people like a baton, but it has six shots and you need to order more ammo for it. You can only carry so much ammo, after all.

An energy weapon, on the other hand. You can recharge it in many places across the station, you can even drag around a recharging unit and a wrench if you are really dedicated. Not to mention, you have the rapid fire power of a laser, but you can also subdue people with the lovely taser.

I don't object to this change, because it's kinda meh, however.

**Comment by DaveTheHeadcrab** [comment]
@tigercat2000 
A gun can only hold so much charge, you can't carry *more* charge.

**Comment by tigercat2000** [comment]
@DaveTheHeadcrab I do think that someone with one of these guns again someone with a revolver would probably have a relatively fair fight however.

**Comment by DaveTheHeadcrab** [comment]
@tigercat2000 
Only because it's a contest of who can stun who first.

**Comment by Plotron** [comment]
I always found the revolver to be more reliable than the tazer when it comes to stunfights.
Doesn't the revolver shoot faster?
And really, HoS e-gun was shit compared to the (now) old revolver as you could easily hoard 4x6 bullets (loaded gun + 3 speedloaders in the bag), for a maximum of 24x15 damage without having to visit... anybody, in fact. That's *360 damage*, good enough for 3 lethal takedowns.

**Comment by Regen1** [comment]
@TheDZD @Fox-McCloud @Markolie 

Any update on this?

**Comment by Fox-McCloud** [comment]
After talking this over with DZD, there's some concerns about this being a tase and lase weapon--on the forums, things are a bit more mixed, with a lot heavily favoring the BS keeping a ballistic weapon, while a few seem ok with this.

I don't have a huge stake in this race, so to speak, but I do have a bit of concerns with Tase and Lase being available to the BS, and I do admit there's a bit of a romantic notion with the revolver--especially now that its damage is negligible.

**Comment by Spacemanspark** [comment]
Quite honestly the Blueshield can ask the captain for an Egun and they'll get one 9/10 times anyways (Cue salt about two separate Blueshields getting the Ion Rifle and targeting me over petty things.).

**Comment by Earthdivine** [comment]
To be honest, the idea of a Tase and Lase doesn't really irk me because it's only 1 of 2 on station. At most I am more worried having access to a Stun weapon would encourage the SecuriShield we tried to do away with.

**Comment by Markolie** [comment]
I'd personally prefer a ballistic weapon of some kind, like the M1911 as suggested on the forums.

**Comment by Spacemanspark** [comment]
Blueshield's start with a taser, as an outside note. That should be removed if this goes forward.

**Comment by Fox-McCloud** [comment]
@Spacemanspark 

https://github.com/ParadiseSS13/Paradise/pull/3157/files#diff-c4b7cb97240ea8c489a6f99ea58ed406L179

File: code/game/jobs/job/supervisor.dm [hunk 3]
```diff
@@ -176,8 +176,13 @@ var/datum/announcement/minor/captain_announcement = new(do_newscast = 1)
 		H.equip_or_collect(new /obj/item/clothing/glasses/hud/health/health_advanced, slot_glasses)
 		H.equip_or_collect(new /obj/item/clothing/under/rank/blueshield(H), slot_w_uniform)
 		H.equip_or_collect(new /obj/item/clothing/suit/armor/vest/blueshield(H), slot_wear_suit)
-		H.equip_or_collect(new /obj/item/weapon/gun/energy/advtaser(H), slot_s_store)
 		H.equip_or_collect(new /obj/item/device/pda/heads/blueshield(H), slot_wear_pda)
+
+		if(H.backbag == 1)
+			H.equip_or_collect(new /obj/item/weapon/gun/energy/blueshield(H), slot_l_hand)
```
Comment (tigercat2000) [inline code/game/jobs/job/supervisor.dm]: shouldn't this just go into the check below

**Comment by Spacemanspark** [comment]
Pardon me while I shut the fuck up.

**Comment by DaveTheHeadcrab** [comment]
I absolutely do not care one way or the other if this gets merged, this was simply a response to people bitching and moaning that,as a blueshield, they didn't get anything lethal with the revolve nerf, however, consider the following:

1: The old revolver, as I mentioned earlier, was basically a one-shot wonder tase'n'lase that also did a damage type objectively more harmful than a laser, *and* with the very real potential to store more shots than any laser gun; This was considered balance, yet concerns are being raised about the potential balance issues of being able to tase'n'lase with this weapon.

2: When people say they want a ballistic weapon for the blueshield I assure you they almost certainly mean they want a *lethal* ballistic weapon for the blueshield in the same vein as the old revolver. As I've said, this is objectively _more_ powerful than the tase'n'lase combo with concerns being raised about it in this PR, so it doesn't take a very large logical leap to see why that's a bad idea.

_OR_

They're worried about not being able to solo nukeops.
As the blueshield your job is not to solo nukeops.
Get over it.

3: This discourages bluecurity by merging their lethal and non-lethal weapon into one, effectively forcing the blueshield to budget his shots until he can get to a recharger.

**Comment by Plotron** [comment]
Tase and lase shouldn't be a problem considering that the pre-nerf revolver did both without having to switch any modes at all and without wasting any ammo.
T&L is available to officers in red alert. An argument's been made that the BS shouldn't have to visit Armory to get gear to do his job in emergencies.

**Comment by DaveTheHeadcrab** [comment]
@Fox-McCloud 
@Markolie 
@TheDZD

**Comment by TheDZD** [comment]
I have no issues with this being merged, although I honestly think it would have been fine to just replace the revolver with an energy gun.

**Comment by DaveTheHeadcrab** [comment]
@TheDZD 
I wholeheartedly agree. But I guarantee you I would never hear the end of it from BS players if I tried that.

**Comment by Fox-McCloud** [comment]
I'm personally find with him having the revolver, especially now that it does a lot less damage per shot----but if he is to have his revolver replaced, then yes--definitely energy gun.

**Comment by HereticChurch** [comment]
Revolver being replaced with this is fine with me. Something to make the Blueshield a little more unique is nice, rather than tossing them a mishmash of items other jobs get. This will leave them with room for one more thing in their line up, as well, since the revolver and taser were sorta merged here.
If people flip out too much on this change its not hard to revert. I doubt they will though, fun unique things like this tend to work out.

**Comment by Plotron** [comment]
@DaveTheHeadcrab, regarding your edited post:
Blueshield in a nuke ops situation is tasked with protecting the head carrying the NAD. In this case it is indeed about fighting nuke ops against all odds, sometimes outnumbered in a dark hallway.
What is more, if the said head goes down with the disk, it'd be Blueshield's job to recover the head (and the disk if possible).

Blueshield in other cases becomes a meatshield of sorts. Dish out as many stuns and damage as you can to save the head with the disk, buying enough time for a succesful evacuation.

You have raised a valid point about *ballistics* some people would want in place of the revolver.
I think this Advanced Stun Revolver is a good compromise.

**Comment by FlattestGuitar** [comment]
[Strawpoll](http://strawpoll.me/6534140/r) I made yesterday with some people voting.

Most agree that Blue should have a reskinned HOSgun with the m1911 coming in at a close second. I say we just make the advanced revolver a thing and see if anyone complains. Really makes no sense to make this drag on any longer.

**Comment by Deanthelis** [comment]
It's such a small feature that I don't think it's unreasonable to just merge it and give it a week to see how it affects the various gamemodes, then leave it or remove it at that point. Frankly, people whining about something like this, in favor or against it, isn't really an accurate way to judge whether or not a feature is actually balanced, since both sides of the argument are horrendously biased. Better to observe actual play, in my opinion.

**Comment by Earthdivine** [comment]
I have to agree that this talk of hypothetical balance and becoming a reskinned HoS talk is taking a bit too long. We have a revert function, just see how it plays and revert it if there's problems.

**Comment by Plotron** [comment]
Revert is used when you encounter serious bugs.

On a live server people just get used to the changes.

It's not that big of a change; the balance will likely not be swayed.

**Comment by DaveTheHeadcrab** [comment]
9 days, folks.

Patently absurd for something that, frankly, doesn't change Blueshield much at all from his state pre-revolver nerf _and_ has seen practically nothing but support on the PR. DZD said on this very thread he's OK with it getting merged.

What is the holdup?

@Fox-McCloud 
@Markolie 
@TheDZD

**Comment by tigercat2000** [comment]
I've had PR's wait for months! Patience, darling. :cat:

File: code/game/jobs/job/supervisor.dm (hunk 3 above)
Comment (Fox-McCloud) [inline code/game/jobs/job/supervisor.dm]: @DaveTheHeadcrab 

This all need to be moved to down below, like @tigercat2000 stated.

**Comment by TheDZD** [comment]
Boop.
